src/Storages/MergeTree/MergeTreeIndexGin.cpp

```diff
@@ -758,44 +763,100 @@ MergeTreeIndexConditionPtr MergeTreeIndexGin::createIndexCondition(const Actions
     return std::make_shared<MergeTreeIndexConditionGin>(predicate, context, index.sample_block, gin_filter_params, token_extractor.get());
 }
 
-MergeTreeIndexPtr ginIndexCreator(const IndexDescription & index)
+namespace
+{
+[[nodiscard]] std::unordered_map<String, Field> convert_arguments_to_options_map(const FieldVector & arguments)
```

Comment:
cosmetic: `[[nodiscard]]` isn't wrong but it would more fit into some library function (e.g. in libcxx), let's remove it.